doc/tutorial.md

```diff
@@ -0,0 +1,192 @@
+```clojure
+user=> (require '[manifold.deferred :as d])
+nil
+user=> (d/chain (aero/get-single c "index" "set-name")
+  #_=>          #(if %1 (prn "good!") (prn "not there")))
+"not there"
+<< nil >>
+```
+If the record existed we would get an `AerospikeRecord`:
+```clojure
+user=> (aero/put c "index" "set-name" 42 1000)
+<< … >>
+user=> (def f (aero/get-single c "index" "set-name"))
+#'user/f
+user=> (deref f)
+#aerospike_clj.client.AerospikeRecord{:payload 42, :gen 1, :ttl 284805805}
+user=>
+```
+We observe a few new things:
+1. The `put` API also return deferred objects. Those will contain the action result when it completes.
+2. The `put` API requires an additional argument, the TTL. Although Aerospike cluster namespaces have a default TTL, and there is also a default TTL in the Java client, we chose to make your life harder here. This is because we believe that TTLs are too important to be default, and programmers must think about them.
+3. In the query result we got a Clojure record with a triplet: `:payload` with the value, `:gen` for the record generation, and a `:ttl` for the record TTL, in Aerospike format. This resembles the [Record class](https://www.aerospike.com/apidocs/java/com/aerospike/client/Record.html). In order to convert the TTL to a Unix EPOCH you can call `expiry-unix` on it (see below).
+#### Unix EPOCH TTL
+Aerospike returns a TTL on the queried records that is Epoch style, but with a different "beginning of time" which is "2010-01-01T00:00:00Z". Call `expiry-unix` with the returned TTL to get a UNIX TTL if you want to convert it later to a more standard timestamp.
+
+```clojure
+user=> (-> f
+  #_=>     deref
+  #_=>     :ttl
+  #_=>     aero/expiry-unix
+  #_=>     java.time.Instant/ofEpochSecond
+  #_=>     str)
+"2019-01-10T08:43:25Z"
+```
+Let's do it in an asynchronous manner:
+```clojure
+user=> (d/chain (aero/get-single c "index" "set-name")
+  #_=>          :ttl
+  #_=>          aero/expiry-unix
+  #_=>          #(java.time.Instant/ofEpochSecond %)
+  #_=>          str
+  #_=>          println)
+<< … >>
+2019-01-10T09:02:45Z
+```
+We got a deferred back and some time later the whole chain of composed logic was triggered.
+
+We can also get the result once multiple required records return. We will simply get a sequence of `AerospikeRecord`s once all of them arrive:
+```clojure
+user=> (run! #(aero/put c (str %1) "set-name" %1 1000) (range 5))
+nil
+user=> @(d/chain (aero/get-multiple c (map str (range 5)) (repeat "set-name"))
+  #_=>           #(map :payload %))
+(0 1 2 3 4)
+```
+
+
+##### Sync Querying
+Since the returned future objects can be easily `deref`ed, simply adding a `@` before queries makes them synchronous.
+
+#### Using Transcoders
+The library takes advantage of deferreds ability to compose and allows you to configure a `:transcoder` to conveniently set this logic:
+* `get` Transcoders are functions of the **AerospikeRecord instance**, not the `deferred` value of it.
```

Comment:
capitalization here is inconsistent with the second bullet

Reply:
ack